core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -83,22 +87,23 @@ protected Collection<String> selectedColumns() {
     return context.selectedColumns();
   }
 
+  protected ExecutorService planExecutor() {
+    return context.planExecutor();
+  }
+
   protected Map<String, String> options() {
     return context.options();
   }
 
-  protected  TableScanContext context() {
+  protected TableScanContext context() {
     return context;
   }
 
   @SuppressWarnings("checkstyle:HiddenField")
-  protected abstract TableScan newRefinedScan(
-      TableOperations ops, Table table, Schema schema, TableScanContext context);
+  protected abstract TableScan newRefinedScan(TableOperations ops, Table table, Schema schema,
+                                              TableScanContext context);
 
-  @SuppressWarnings("checkstyle:HiddenField")
-  protected abstract CloseableIterable<FileScanTask> planFiles(
```

Comment:
I am not sure the signature of this method makes sense after we introduced the context object. This method passes some arguments but not all so the implementations are not entirely pure. Also, the passed arguments are useless for some metadata tables. That's why I see little value in having so many arguments. Just makes calling this method harder.